MenuOOPs3/src/ShoppingStore/Bill.java

```diff
@@ -0,0 +1,134 @@
+package ShoppingStore;
+
+import java.util.ArrayList;
+
+/**
+ * Display bill on console and write on text file
+ * 
+ * @author Udit Saxena
+ * 
+ */
+public class Bill {
+	private String userName;
+	private Cart cart;
+	private ProductPromo productPromo;
+
+	/**
+	 * parameterized constructor initializing private variables
+	 * 
+	 * @param userName
+	 *            - User Name
+	 * @param cart
+	 *            - Cart class object
+	 * @param productPromo
+	 *            - ProductPromo class object
+	 */
+	public Bill(String userName, Cart cart, ProductPromo productPromo) {
+		this.userName = userName;
+		this.cart = cart;
+		this.productPromo = productPromo;
+	}
+
+	/**
+	 * 
+	 * @return String containing purchase details
+	 */
+	private String purchaseProductDetails() {
+		String output = "";
+		for (CartProducts product : cart.getCartItems()) {
+			output += "\nProduct : " + product.getProductId() + " - "
+					+ product.getProductName() + "\n";
+			output += "Cost : Rs "
+					+ (product.getProductPrice() * product.getQuantity()) + "("
+					+ product.getQuantity() + " X " + product.getProductPrice()
+					+ ")\n";
+		}
+
+		return output;
+	}
+
+	/**
+	 * 
+	 * @return String containing Promotion/Discount details
+	 */
+	private String appliedPromotionsOrDiscount() {
+		String output = "";
```

Comment:
This logic is same logic we have used when we apply discount on products, same can be stored in cartProduct and used here to display only.